theano/sandbox/cuda/opt.py

```diff
@@ -97,6 +97,7 @@
 register_opt(name='gpu_constant_folding')(
     tensor.opt.constant_folding)
 register_opt()(theano.tensor.opt.local_subtensor_make_vector)
+register_opt('unsafe')(theano.tensor.opt.local_remove_all_assert)
```

Comment:
I've added the following: 
```python
gpu_optimizer.register('local_remove_all_assert', 
         theano.tensor.opt.local_remove_all_assert, 
         'unsafe')
```
since adding `'gpu'` tag caused `unsafe` by default. And added corresponding tests for cuda/gpuarray